doc/source/whatsnew/v0.21.1.txt

```diff
@@ -56,6 +56,8 @@ Documentation Changes
 
 Bug Fixes
 ~~~~~~~~~
+- Bug in :meth:`fillna` where maximum recursion depth gets exceeded in comparison (:issue:`18159`).
```

Comment:
make this friendly from a user perspective, eg. a reader of the release notes should understand that doing X is now fixed.